vizro-core/tests/unit/vizro/models/_controls/conftest.py

```diff
@@ -17,3 +17,22 @@ def managers_one_page_two_graphs(gapminder):
         ],
     )
     Vizro._pre_build()
+
+
+@pytest.fixture
+def managers_one_page_container_controls(gapminder):
+    """Instantiates a simple model_manager and data_manager with a page, and two graph models and gapminder data."""
+    vm.Page(
+        id="test_container",
+        title="My first dashboard",
+        components=[
+            vm.Container(
+                title="",
+                components=[
+                    vm.Graph(id="scatter_chart", figure=px.scatter(gapminder, x="lifeExp", y="gdpPercap")),
+                ],
```

Comment:
What do you think about this approach:
```suggestion
                components=[
                    vm.Graph(id="scatter_chart", figure=px.scatter(gapminder, x="lifeExp", y="gdpPercap")),
                ],
                controls=[
                    vm.Filter(id="container_filter", column="continent", selector=vm.Checklist(value=["Europe"])),
                    vm.Parameter(
                        id="container_parameter",
                        targets=["scatter_chart.x"],
                        selector=vm.Checklist(options=["lifeExp", "gdpPercap", "pop"], value=["lifeExp"]),
                    )
                ]
```

In tests you can just `filter=model_manager["container_filter"]` and `assert`. Similar for Parameter.

Also, you can remove `Vizro._pre_build()` from here and then explicitly call it from the tests.